### ISSUE TITLE
[Bug]: Hybrid search failed with error: Can't get bitset element at 102 err %!w(<nil>))>

### ISSUE THREAD

**Issue by alexandervyatsu** [milvus-io/milvus#20534]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.1.4 deb
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

2.5M 1024-dim vectors
index: IVF_PQ, nlist=1024, m=256
8 scalar fields int16 (b0,b1,..,b7)

Hybrid search with expr="b0 in [1,2,3,4,5]" fails
with RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err =fail to Search, QueryNode ID=1, reason=Search 1 failed, reason [UnexpectedError] Assert "get_bit(field_data_ready_bitset_, field_id)" at /root/milvusdeb-2.1/milvus/internal/core/src/segcore/SegmentSealedImpl.cpp :298
  => Can't get bitset element at 102 err %!w(<nil>))>, <Time:{'RPC start': '2022-11-12 11:58:56.704749', 'RPC error': ' 2022-11-12 11:58:56.724746'}>
  
Hybrid search with expr="b0 == 1 or b0 == 2 or b0 == 3" or expr="b0 < 6" works without errors. 

The error does not appear on a collection of 100K vectors.

**Comment by xiaofan-luan** [comment]
/assign @cydrain 
pls help on investigating

**Comment by xiaofan-luan** [comment]
/assign @longjiquan 
pls help on investigating it

**Comment by longjiquan** [comment]
hello, @alexandervyatsu , can you provide the log of Milvus server?

**Comment by longjiquan** [comment]
Have you retried "b0 in [1,2,3,4,5]" after the error is raised? @alexandervyatsu

**Comment by longjiquan** [comment]
I guess the collection is not fully loaded when you got this error, so after the collection is fully loaded, other scalar filtering succeeded.

**Comment by alexandervyatsu** [comment]
The collection has been fully loaded.
b0 in [1, 2, 3, 4, 5 ] didn't work after b0 < 6 worked successfully.

**Comment by alexandervyatsu** [comment]
> I guess the collection is not fully loaded when you got this error, so after the collection is fully loaded, other scalar filtering succeeded.

This error occurs when building an index on a scalar field.
Everything works without an index.

**Comment by xiaofan-luan** [comment]
@longjiquan any update on those related issues?

**Comment by longjiquan** [comment]
@alexandervyatsu thx for the very detailed information. This issue has already been fixed in #20840 in 2.2.0 branch, and this fix will be released in our `2.2.1` version. Maybe you can retry this later in `v2.2.1`, besides this, `v2.2.1` also resolved many issues, such as memory, performance, failure recovery and etc.

**Comment by alexandervyatsu** [comment]
Thanks for the quick bug fix.

**Comment by longjiquan** [comment]
Already fixed in #20840 .

/close
